### PR TITLE
Remember selected input devices

### PR DESCRIPTION
**Pull request by danxuliu**
Follow up to #3913

Before the selected input devices were used as long as the device was still available and Talk was still open. However, if the device was disconnected and connected it was not automatically selected again, even if Talk was still open. Now the last selected device of each kind is remembered, so it will be automatically used if available when joining a call, even if it was disconnected and connected or if Talk was closed and opened.
    
Additionally, if audio or video devices were explicitly disabled this will also be remembered and respected when Talk is open again.

Note that the last selected device is automatically used if available when joining a call, but it will not be automatically used if it becomes available during a call. While it would be possible to do that at least for me it felt strange and unexpected (but I would have no problem enabling that if desired).

Besides that this pull request also fixes some related issues that have given me a big headache :-P

### CODE REVIEW

File: src/utils/webrtc/MediaDevicesManager.js [hunk 1]
```diff
@@ -83,6 +91,15 @@ export default function MediaDevicesManager() {
 	this._tracks = []
 
 	this._updateDevicesBound = this._updateDevices.bind(this)
+
+	this._pendingEnumerateDevicesPromise = null
+
+	if (localStorage.getItem('audioInputId') === LOCAL_STORAGE_NULL_DEVICE_ID) {
```
Comment (nickvergessen) [inline src/utils/webrtc/MediaDevicesManager.js:97]: Please use the storage from `../../services/BrowserStorage.js`

**Review by nickvergessen — CHANGES_REQUESTED** [review]
Works good in general, but I think the null device needs a bit more handling in this case. When you ever selected "No camera", you leave the call and join again, you will not be prompted for video permissions and your button will always say "No camera". Maybe clicking that button could instead bring you to the settings so you can select one?

File: src/utils/webrtc/MediaDevicesManager.js (hunk 1 above)
Comment (danxuliu) [reply src/utils/webrtc/MediaDevicesManager.js]: Ugh, right :facepalm: Done.

**Comment by danxuliu** [comment]
> Works good in general, but I think the null device needs a bit more handling in this case. When you ever selected "No camera", you leave the call and join again, you will not be prompted for video permissions and your button will always say "No camera". Maybe clicking that button could instead bring you to the settings so you can select one?

Yes, sounds good. I would do it in a follow up pull request, though, as there are currently some issues when getting the media fails that should probably be fixed first (as that may affect the code to differentiate between media explicitly disabled and media failed).

**Review by nickvergessen — APPROVED** [review]
:shark:

**Pull request merged**